packages/finch-graphql/src/client/FinchClient.ts

```diff
@@ -0,0 +1,54 @@
+import { DocumentNode } from 'graphql';
+import gql from 'graphql-tag';
+import { QueryCache, Listener } from './cache';
+import { FinchQueryOptions, GenericVariables } from '../types';
+import { isDocumentNode } from '../utils';
+import { queryApi } from './client';
+
+interface FinchClientOptions {
+  cache?: QueryCache;
+  id?: string;
+  messageKey?: string;
+}
+
+export class FinchClient {
+  private cache: QueryCache | undefined;
```

Comment:
Let's make a generic type for cache when QueryCache can implement. That way if we want to pass custom caching modules it should be super simple.